eng/pipelines/docs.yml

```diff
@@ -2,85 +2,103 @@ trigger:
   - master
 jobs:
   - job: CreateDocs
-    variables:
-      skipComponentGovernanceDetection: true
     pool:
       vmImage: vs2017-win2016
     steps:
+      # First set of steps download and install tools or create directories to be used in doc generation
+      - powershell: |
+          Invoke-WebRequest -Uri "https://github.com/mono/api-doc-tools/releases/download/mdoc-5.7.4.9/mdoc-5.7.4.9.zip" `
+          -OutFile "mdoc.zip" | Wait-Process; Expand-Archive -Path "mdoc.zip" -DestinationPath "./mdoc/"
+        workingDirectory: $(Build.BinariesDirectory)
+        displayName: Download and Extract mdoc Zip
+      - task: NuGetToolInstaller@0
+        displayName: Install Nuget Client
+      - powershell: nuget install Microsoft.DocAsCode.ECMA2Yaml -ExcludeVersion
+        workingDirectory: $(Build.BinariesDirectory)
+        displayName: Install Microsoft.DocAsCode.ECMA2Yaml
       - powershell: |
           Invoke-WebRequest -Uri "https://github.com/dotnet/docfx/releases/download/v2.43.2/docfx.zip" `
           -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "./docfx/"
         workingDirectory: $(Build.BinariesDirectory)
         displayName: Download and Extract DocFX
-      - task: CmdLine@2
-        displayName: Provision DocFX Directory
+      - powershell: ". '$(Build.SourcesDirectory)/eng/docgeneration/script/docgen.ps1';CreateDocGenFolders"
```

Comment:
I would do this as inline ps code. No need to source in the script and then call the function especially for something that isn't really sharing any logic or state.

Reply:
Done